generative/maisi/README.md

```diff
@@ -0,0 +1,122 @@
+```bash
+python inference.py -c ./configs/config_maisi.json -e ./configs/environment.json --num 1
+```
+`--num` defines how many images it would generate.
+
+An example output is shown below.
+<p align="center">
+  <img src="./figs/syn_axial.png" width="30%" >
+&nbsp; &nbsp; &nbsp; &nbsp;
+  <img src="./figs/syn_sag.png" width="30%" >
+&nbsp; &nbsp; &nbsp; &nbsp;
+  <img src="./figs/syn_cor.png" width="30%" >
+</p>
+
+### 3. Training example
+Training data preparation can be found in [./data/README.md](./data/README.md)
+
+#### [3.1 3D Autoencoder Training](./train_autoencoder.py)
+
+The network configuration files are located in [./config/config_train_32g.json](./config/config_train_32g.json) for 32G GPU
+and [./config/config_train_16g.json](./config/config_train_16g.json) for 16G GPU.
+You can modify the hyperparameters in these files to suit your requirements.
+
+The training script resamples the brain images based on the voxel spacing specified in the `"spacing"` field of the configuration files. For instance, `"spacing": [1.1, 1.1, 1.1]` resamples the images to a resolution of 1.1x1.1x1.1 mm. If you have a GPU with larger memory, you may consider changing the `"spacing"` field to `"spacing": [1.0, 1.0, 1.0]`.
+
+The training script uses the batch size and patch size defined in the configuration files. If you have a different GPU memory size, you should adjust the `"batch_size"` and `"patch_size"` parameters in the `"autoencoder_train"` to match your GPU. Note that the `"patch_size"` needs to be divisible by 4.
+
+Before you start training, please set the path in [./config/environment.json](./config/environment.json).
+
+- `"model_dir"`: where it saves the trained models
+- `"tfevent_path"`: where it saves the tensorboard events
+- `"output_dir"`: where you store the generated images during inference.
+- `"resume_ckpt"`: whether to resume training from existing checkpoints.
+- `"data_base_dir"`: where you store the Brats dataset.
+
+Below is the the training command for single GPU.
+
+```bash
+python train_autoencoder.py -c ./config/config_train_32g.json -e ./config/environment.json -g 1
+```
+
+The training script also enables multi-GPU training. For instance, if you are using eight 32G GPUs, you can run the training script with the following command:
+```bash
+export NUM_GPUS_PER_NODE=8
+torchrun \
+    --nproc_per_node=${NUM_GPUS_PER_NODE} \
+    --nnodes=1 \
+    --master_addr=localhost --master_port=1234 \
+    train_autoencoder.py -c ./config/config_train_32g.json -e ./config/environment.json -g ${NUM_GPUS_PER_NODE}
+```
+
+<p align="center">
+  <img src="./figs/train_recon.png" alt="autoencoder train curve" width="45%" >
+&nbsp; &nbsp; &nbsp; &nbsp;
+  <img src="./figs/val_recon.png" alt="autoencoder validation curve" width="45%" >
+</p>
+
+With eight DGX1V 32G GPUs, it took around 55 hours to train 1000 epochs.
+
+#### [3.2 3D Latent Diffusion Training](./train_diffusion.py)
```

Comment:
It would be great if you could also mention the memory usage and training time in 3.2 and 3.3

Reply:
Will be added by Pengfei and Dong in the next few PRs